src/block-components/button/style.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * External dependencies
+ */
+import { Style as StyleComponent } from '~stackable/components'
+import {
+	useStyles, getStyles,
+} from '~stackable/util'
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element'
+import { sprintf } from '@wordpress/i18n'
+
+/**
+ * Internal dependencies
+ */
+import { BorderStyle } from '../helpers/borders'
+
+const getStyleParams = options => {
+	const {
+		selector,
+		textSelector,
+		textHoverSelector,
+	} = options
+
+	return [
+		{
+			selector,
+			responsive: 'all',
+			styleRule: 'minHeight',
+			attrName: 'buttonMinHeight',
+			format: '%spx',
+		},
+		{
+			selector,
+			responsive: 'all',
+			styleRule: 'paddingTop',
+			attrName: 'buttonPadding',
+			hasUnits: 'px',
+			valuePreCallback: value => value?.top,
+		},
+		{
+			selector,
+			responsive: 'all',
+			styleRule: 'paddingRight',
+			attrName: 'buttonPadding',
+			hasUnits: 'px',
+			valuePreCallback: value => value?.right,
+		},
+		{
+			selector,
+			responsive: 'all',
+			styleRule: 'paddingBottom',
+			attrName: 'buttonPadding',
+			hasUnits: 'px',
+			valuePreCallback: value => value?.bottom,
+		},
+		{
+			selector,
+			responsive: 'all',
+			styleRule: 'paddingLeft',
+			attrName: 'buttonPadding',
+			hasUnits: 'px',
+			valuePreCallback: value => value?.left,
+		},
+		{
+			selector,
+			styleRule: 'background',
+			attrName: 'buttonBackgroundColor',
+			valueCallback: ( _, getAttribute ) => {
+				const buttonBackgroundGradientDirection = getAttribute( 'buttonBackgroundGradientDirection' )
+				const buttonBackgroundColor = getAttribute( 'buttonBackgroundColor' )
+				const buttonBackgroundColor2 = getAttribute( 'buttonBackgroundColor2' )
+
+				if ( getAttribute( 'buttonBackgroundColorType' ) !== 'gradient' ) {
+					return getAttribute( 'buttonBackgroundColor' )
+				}
+
+				return `linear-gradient(${ buttonBackgroundGradientDirection !== '' ? buttonBackgroundGradientDirection + 'deg' : '90deg' }, ${ buttonBackgroundColor || buttonBackgroundColor2 }, ${ buttonBackgroundColor2 || buttonBackgroundColor })`
+			},
+			dependencies: [ 'buttonBackgroundGradientDirection', 'buttonBackgroundColor', 'buttonBackgroundColor2', 'buttonBackgroundColorType' ],
+		},
+		{
+			selector: `${ selector }:after`,
+			styleRule: 'background',
+			attrName: 'buttonBackgroundColor',
+			hover: 'all',
```

Comment:
The hover is incorrectly added to the block instead of the button. The `hoverSelector` should be added:

```suggestion
			hover: 'all',
			hoverSelector: `${ selector }:hover:after`,
```

Reply:
Line 116 is responsible for the button hover. This line only adds the overlay gradient which will appear after when hovered.